### PR TITLE
Fix spamming buttons on main menu

### PR DESCRIPTION
**Pull request by m4rtint**

### CODE REVIEW

File: Assets/Script/UI/MainMenu/StartMenuScreen.cs [hunk 1]
```diff
@@ -31,7 +31,10 @@ public class StartMenuScreen : MainMenuScreens
     /// </summary>
     protected override void Awake()
     {
-        m_startGame.onClick.AddListener(StartGoingToNextScreen);
+        m_startGame.onClick.AddListener(() => {
+            SetButtonInteractable(false);
+            StartGoingToNextScreen();
```
Comment (m4rtint) [inline Assets/Script/UI/MainMenu/StartMenuScreen.cs]: If you spam the confirm button too fast, it'lll skip straight to the game, 
So i disable the button when you press it.

File: Assets/Script/UI/MainMenu/StartMenuScreen.cs [hunk 2]
```diff
@@ -40,6 +43,11 @@ protected override void Awake()
         m_quitPanel.OnClosePanel += OnCloseQuitPanel;
     }
 
+    private void OnEnable()
+    {
+        SetButtonInteractable(true);
```
Comment (m4rtint) [inline Assets/Script/UI/MainMenu/StartMenuScreen.cs]: Enable it when you open it again

File: Assets/Script/UI/MainMenu/BackButton.cs [hunk 3]
```diff
@@ -22,7 +22,7 @@ public class BackButton : MonoBehaviour
     [SerializeField] private float m_durationToHold = 3.0f;
     [SerializeField] private Image m_loadingBackButton;
     [SerializeField] private TextMeshProUGUI m_playerUsingBackButton;
-    private float m_currentTimer;
+    private float m_currentTimer = float.MaxValue;
```
Comment (m4rtint) [inline Assets/Script/UI/MainMenu/BackButton.cs]: This was set as 0, so if you press b when you first set this as active it, it'll instantly go back.

Comment (mikeslgoh) [reply Assets/Script/UI/MainMenu/BackButton.cs]: nice catch

**Comment by mikeslgoh** [comment]
[Spam Buttons in main menu causes multiple screens](https://trello.com/c/61MMecsB/246-spam-buttons-in-main-menu-causes-multiple-screens)

**Pull request merged**